File: mall-goods-interface/src/main/java/com/mall/goods/service/SellerService.java
package com.mall.goods.service;

import com.mall.pojo.Seller;
import common.pojo.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 商家/卖家服务接口
 * @author Wwl
 *
 */
public interface SellerService {

	/**
	 * 增加
	 * @param seller
	 */
	public void add(Seller seller);

	/**
	 * 修改
	 * @param seller
	 */
	public void update(Seller seller);

	/**
	 * 根据Id获取实体
	 * @param sellerId
	 * */
	public Seller findOne(String sellerId);

	/**
	 * 查询所有
	 * @return
	 */
	public List<Seller> findAll();

	/**
	 * 查询所有-分页
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);

	/**
	 * 根据条件查询，分页
	 * @param seller
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Seller seller, int pageNum, int pageSize);

	/**
	 * 删除
	 * @return
	 */
	public void delete(String[] ids);

}

File: mall-shop-web/src/main/webapp/js/service/sellerService.js
/**
 * 卖家服务层
 * Created by Wwl on 2019/4/18
 */
app.service('sellerService', function($http){

    //增加
    this.add = (entity) => $http.post('../seller/add.do',entity);

    //修改
    this.update = (entity) => $http.post('../seller/update.do',entity);

    //删除
    this.delete = (ids) => $http.get('../seller/delete.do?ids='+ids);

    //查询实体
    this.findOne = (id) => $http.get('../seller/findOne.do?id='+id);

    //查询列表
    this.findList = () => $http.get('../seller/list.do');

    //分页查询
    this.findPage = (page,size) => $http.get('../seller/listByPage.do?page=' + page + '&size=' + size);

    //搜索
    this.search = (page,size,searchEntity) => $http.post('../seller/search.do?page='+page+'&size='+size,searchEntity);

});